src/include/schema.h

```diff
@@ -256,6 +256,25 @@ struct __wt_table {
 	}								\
 } while (0)
 
+/*
+ * WT_WITH_HOTBACKUP_LOCK --
+ * 	Acquire the hot backup lock, perform an operation, drop the lock.
+ */
+#define	WT_WITH_HOTBACKUP_LOCK(session, op) do {		\
+	WT_CONNECTION_IMPL *__conn = S2C(session);			\
+	if (F_ISSET(session, WT_SESSION_HOTBACKUP_LOCKED)) {		\
+		op;							\
+	} else {							\
+		__wt_readlock(session, &__conn->hot_backup_lock);	\
+		F_SET(session, WT_SESSION_HOTBACKUP_LOCKED);		\
+		do {							\
+			op;						\
```

Comment:
I've wrapped this in a loop so that when we decide to jump out of the code block with `break`, we unlock cleanly.